File: src/games/individual/whichlarger.ts
import { GameSession, AiResponse } from '../../types';
import { BaseGame } from '../base';
import { AiService } from '../../ai/api';
import { SessionManager } from '../../session/manager';
import { PointsSystem } from '../../points/system';
import { GroupManager } from '../../group/manager';
import { LeaderboardManager } from '../../leaderboard/manager';
import { KvStore } from '../../db/kv';
import { DbStore } from '../../db/d1';
import { TelegramApi } from '../../telegram';
import { Env } from '../../types';

export class WhichLargerGame extends BaseGame {
  constructor(
    ai: AiService, sessionManager: SessionManager, pointsSystem: PointsSystem,
    groupManager: GroupManager, leaderboardManager: LeaderboardManager,
    kv: KvStore, db: DbStore, telegram: TelegramApi, env: Env
  ) { super(ai, sessionManager, pointsSystem, groupManager, leaderboardManager, kv, db, telegram, env); }

  getGameId(): string { return 'whichlarger'; }

  async generateQuestion(session: GameSession, round: number): Promise<AiResponse> {
    const num1 = Math.floor(Math.random() * 1000);
    const num2 = Math.floor(Math.random() * 1000);
    const correct = num1 >= num2 ? '1' : '2';
    return {
      success: true, question: 'أيهما أكبر؟', options: [String(num1), String(num2)], correctAnswer: correct,
      storyBranch: [], choices: [], hint: '', explanation: '',
      metadata: { num1, num2 },
    };
  }

  async checkAnswer(session: GameSession, userId: number, answer: string): Promise<boolean> {
    const q = session.currentQuestion;
    if (!q) return false;
    return answer.trim() === q.correctAnswer;
  }

  renderQuestion(session: GameSession, q: AiResponse): string {
    const [num1, num2] = q.options;
    return `🔢 أي رقم أكبر؟\n\n1. ${num1}\n2. ${num2}\n\nأرسل رقم الإجابة:`;
  }

  renderRoundStart(session: GameSession): string {
    return '🎯 لعبة أيهما أكبر!\n\nسيتم عرض رقمين وعليك اختيار الأكبر.';
  }
}

File: python_bot/games/whichlarger.py
#!/usr/bin/env python3
"""أيهما أكبر؟ - اختر الرقم الأكبر"""
import random
from games.base import BaseGame

class WhichLargerGame(BaseGame):
    def get_game_id(self):
        return "whichlarger"
    def generate_question(self, session):
        n1 = random.randint(1, 1000)
        n2 = random.randint(1, 1000)
        correct = "1" if n1 >= n2 else "2"
        return {"question": "أيهما أكبر؟", "options": [str(n1), str(n2)], "correct_answer": correct}
    def check_answer(self, session, user_id, answer):
        q = session['question_data']
        return answer.strip() == q['correct_answer']
    def render_question(self, session, question):
        return f"🔢 أي رقم أكبر؟\n\n1. {question['options'][0]}\n2. {question['options'][1]}\n\nأرسل رقم الإجابة:"